Source/Common/Preferences.cpp

```diff
@@ -264,6 +262,19 @@ int Preferences::Config_Save()
     if (Config(__T("Sponsored")).empty()) Config(__T("Sponsored"))=__T("0");
     if (Config(__T("Theme")).empty()) Config(__T("Theme"))=__T("0");
 
+    // Disable legacy shell extension on Windows 11
+    DWORD dwVersion=GetVersion();
+    DWORD dwMajorVersion=(DWORD)(LOBYTE(LOWORD(dwVersion)));
+    DWORD dwBuild=0;
+    if (dwVersion<0x80000000) dwBuild=(DWORD)(HIWORD(dwVersion));
+    if (dwMajorVersion==10 && dwBuild>=22000) {
+        Config(__T("ShellExtension"))=__T("0");
+        Config(__T("ShellExtension_Folder"))=__T("0");
+    } else {
+        if (Config(__T("ShellExtension")).empty()) Config(__T("ShellExtension"))=__T("1");
+        if (Config(__T("ShellExtension_Folder")).empty()) Config(__T("ShellExtension_Folder"))=__T("0");
+    }
+

```

Comment:
I don't see the added value of not using `ShellExtension` on both Win10- and Win11+.
In my opinion this part should not be changed and other parts of the PR should be adapted to `ShellExtension` value.

Reply:
ShellExtension and ShellExtension_Folder is used multiple times here:
https://github.com/MediaArea/MediaInfo/blob/master/Source/Common/Preferences.cpp

So I did not want to touch that and break the old menu or mess up some registry.

Reply:
> So I did not want to touch that and break the old menu or mess up some registry.

My understanding is that we could live with (nearly) nothing changed in MI GUI so no risk to mess up.
I don't see why it is not possible to do https://github.com/MediaArea/MediaInfo/pull/960#discussion_r1844972395.